[DROOLS-7481] Add Cep complex test : ansible use cases : 2nd round

tkobayas opened the pull request:
- ReliabilityCepOnceWithinTest : ok
- ReliabilityCepTimedOutTest : Had an issue with a nested object reference after deserialization
- Introduce EqualityObjectStore
- Introduce UUID in HashMapFactImpl for equals/hashCode
- WIP

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-7481

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-lts</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

 - for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native-lts</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

<!-- TODO to uncomment if activating the quarkus-3 rewrite PR job -->
<!-- <details>
<summary>
Quarkus-3 PR check is failing ... what to do ?
</summary>
The Quarkus 3 check is applying patches from the `.ci/environments/quarkus-3/patches`.

The first patch, called `0001_before_sh.patch`, is generated from Openrewrite `.ci/environments/quarkus-3/quarkus3.yml` recipe. The patch is created to speed up the check. But it may be that some changes in the PR broke this patch.  
No panic, there is an easy way to regenerate it. You just need to comment on the PR:
```
jenkins rewrite quarkus-3
```
and it should, after some minutes (~20/30min) apply a commit on the PR with the patch regenerated.

Other patches were generated manually. If any of it fails, you will need to manually update it... and push your changes.
</details> -->

kie-ci4 commented:
**(tests) - kogito-springboot-examples job** `#717` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/drools/main/.ci/buildchain-config.yaml' -o 'bc' -p kiegroup/kogito-examples -u https://github.com/kiegroup/drools/pull/5344 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-springboot-examples/717/display/redirect

**Test results:**
- PASSED: 129
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-springboot-examples/717/testReport/org.kie.kogito.springboot.outbox/OutboxIT/testSendProcessEvents/">org.kie.kogito.springboot.outbox.OutboxIT.testSendProcessEvents</a></summary>
expected: <true> but was: <false>
</details>

tkobayas commented on drools-reliability/drools-reliability-infinispan/src/test/java/org/drools/reliability/infinispan/ReliabilityCepTimedOutTest.java:
```diff
@@ -0,0 +1,291 @@
+                                      .expr( after(0, timeAmount.getTimeUnit(), timeAmount.getAmount(), timeAmount.getTimeUnit()), controlVar1 ) ),
+                              protoPattern(controlVar3).expr(p -> ((String)p.get("rulename")).startsWith(rulePrefix)),
+                              on(controlVar1, controlVar3).execute((drools, c1, c3) -> {
+                                  drools.delete(c3.get("event"));
```
If we use `IdentityObjectStore`, after failover, deserialized `HashMapEventImpl` and the same `HashMapEventImpl` which is included in the controlEvent (as "event" in its map) are considered as different objects. So This line cannot find the `HashMapEventImpl` in objectStore, so results in NullPointerException.

Possible solutions:
A) If we serialize/deserialize the whole ObjectStore with the same ObjectOutputStream/ObjectInputStream, the object reference is automatically resolved. (https://docs.oracle.com/javase/6/docs/platform/serialization/spec/output.html)
  => It would be a heavy operation, not realistic.

B) Use `EqualityObjectStore` and implement `UUID` in `HashMapFactImpl` and `equals`/`hashCode` based on the uuid. It helps to find the same fact in objectStore, so the test passed.
  => Remaining concern:
    - They are still different objects. Updating the original event wouldn't update the event inside the controlEvent
    - This solution only for prototype (ansible) use cases. For usual class facts, they should implement `equals`/`hashCode`.

This JIRA implemented B). @mariofusco Do you have any concerns or alternative ideas?

mariofusco replied:
Option A) is clearly not realistic or even reasonable, but in my opinion also B) is only a workaround with potentially huge implications (if you remember we decided to use the `IdentityMap` to avoid relying on potentially poorly written user `equals`/`hashcode` methods) that is attempting to solve the wrong problem.

@tkobayas as discussed I believe that the problem is caused by a wrongly written rule that does something like

  `drools.delete(c3.get("event"));`

instead deleting an event that has been properly retrieved via LHS pattern matching. Following this idea, I successfully changed that rule in the ansible integration codebase with [this commit](https://github.com/kiegroup/drools-ansible-rulebook-integration/commit/711ad015893c0cc1f469a793df886b5415e8d7e9). Please revert the use of that `EqualityMap`, rewrite the rules in your test case according to how I changed them in the ansible integration and check if this is enough to be compatible also with the reliable session.

tkobayas commented on drools-reliability/drools-reliability-core/src/main/java/org/drools/reliability/core/ReliablePseudoClockScheduler.java:
```diff
@@ -65,11 +66,11 @@ private void updateStorage() {
      */
     public List<TimerJobInstance> createFilteredInternalQueueForPersistence(PriorityQueue<TimerJobInstance> queue) {
         return queue.stream()
-                    .filter(job -> !(job.getJob() instanceof ExpireJob))
+                    .filter(job -> !(job.getJob() instanceof ExpireJob || job.getJob() instanceof TimerNodeJob))
```
`TimerNodeJob` (= temporal operators) is also created by repropagate, so we don't need to persist

kie-ci4 commented:
**(tests) - kogito-apps job** `#1486` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/drools/main/.ci/buildchain-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/drools/pull/5344 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/1486/display/redirect

**Test results:**
- PASSED: 2946
- FAILED: 0

Those are the test failures: none

See console log:
<details>
<summary><b>Console Logs</b></summary>
[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: Job Sink Recipient - Parent SUCCESS [  0.024 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: Job Sink Recipient - Runtime SUCCESS [  3.674 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: Job Sink Recipient - Deployment SUCCESS [  1.828 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: Common .............. SUCCESS [02:13 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: PostgreSQL .......... SUCCESS [02:43 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: PostgreSQL .......... SUCCESS [  5.417 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: In-memory ........... SUCCESS [02:20 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs :: Addons ...................... SUCCESS [  0.025 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Jobs Service Embedded Quarkus Addon - Parent ....... SUCCESS [  0.026 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Jobs Service Embedded Quarkus Addon - Runtime ...... SUCCESS [  3.981 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Jobs Service Embedded Quarkus Addon - Deployment ... SUCCESS [  2.758 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: Infinispan .......... SUCCESS [02:31 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Jobs Service :: MongoDB ............. SUCCESS [02:14 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index .......................... SUCCESS [  0.021 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage .................. SUCCESS [  0.018 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage API .............. SUCCESS [  1.124 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage Protobuf ......... SUCCESS [  1.301 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage PostgreSQL ....... SUCCESS [ 19.371 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage Infinispan ....... SUCCESS [  2.925 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage MongoDB .......... SUCCESS [ 15.189 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage PostgreSQL :: Reporting SUCCESS [ 19.199 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Storage Oracle DB ........ SUCCESS [ 53.535 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Common ................... SUCCESS [  3.909 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service .................. SUCCESS [  0.021 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service Common ........... SUCCESS [ 10.957 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service with In-memory persistence SUCCESS [01:46 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service with Infinispan .. SUCCESS [02:47 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service with MongoDB ..... SUCCESS [02:05 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service with Oracle ...... SUCCESS [03:44 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Data Index Service with PostgreSQL .. SUCCESS [01:59 min]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index .... SUCCESS [  0.021 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Common SUCCESS [  0.019 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Common :: Runtime SUCCESS [  2.792 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Common :: Deployment SUCCESS [  0.295 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index In-memory SUCCESS [  0.019 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index In-memory :: Runtime SUCCESS [  0.299 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index In-memory :: Deployment SUCCESS [  0.448 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index In-memory :: Integration Tests :: SW SUCCESS [ 23.837 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index In-memory :: Integration Tests :: Process SUCCESS [ 21.380 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index PostgreSQL SUCCESS [  0.020 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index PostgreSQL :: Runtime SUCCESS [  0.264 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index PostgreSQL :: Deployment SUCCESS [  0.406 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index PostgreSQL :: Integration Tests :: SW SUCCESS [ 26.846 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index PostgreSQL :: Integration Tests :: Process SUCCESS [ 24.451 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Infinispan SUCCESS [  0.019 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Infinispan :: Runtime SUCCESS [  0.344 s]<br/>[2023-06-23T08:47:40.347Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Infinispan :: Deployment SUCCESS [  0.500 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Serverless Workflow Quarkus Dev UI Extension SUCCESS [  0.019 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Serverless Workflow Quarkus Dev UI Extension :: Extension SUCCESS [  2.474 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: UI Packages ......................... SUCCESS [17:32 min]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Serverless Workflow Quarkus Dev UI Extension :: Deployment SUCCESS [ 20.366 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Infinispan :: Integration Tests :: SW SUCCESS [ 37.262 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index Infinispan :: Integration Tests :: Process SUCCESS [ 28.637 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index MongoDB SUCCESS [  0.024 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index MongoDB :: Runtime SUCCESS [  0.360 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index MongoDB :: Deployment SUCCESS [  0.712 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index MongoDB :: Integration Tests :: SW SUCCESS [ 28.483 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Kogito Addons Quarkus Data Index MongoDB :: Integration Tests :: Process SUCCESS [ 20.368 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Security Commons .................... SUCCESS [ 56.770 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Management Console .................. SUCCESS [ 14.161 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty UI ........................... SUCCESS [01:16 min]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Task Console ........................ SUCCESS [ 14.747 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Explainability ...................... SUCCESS [  0.019 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Explainability Api .................. SUCCESS [  1.758 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Explainability Service .............. SUCCESS [  9.682 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty .............................. SUCCESS [  0.018 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Storage ...................... SUCCESS [  0.018 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Storage API .................. SUCCESS [  1.434 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Service ...................... SUCCESS [  0.019 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Service :: API ............... SUCCESS [  0.212 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Explainability Service Messaging .... SUCCESS [ 29.157 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Explainability Service Rest ......... SUCCESS [ 47.490 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Storage Common ............... SUCCESS [  0.172 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Storage Infinispan ........... SUCCESS [  2.519 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Storage Redis ................ SUCCESS [  1.889 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Storage PostgreSQL ........... SUCCESS [  2.420 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Service Common ............... SUCCESS [01:52 min]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Service Infinispan ........... SUCCESS [ 34.959 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Service Redis ................ SUCCESS [ 26.795 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Trusty Service PostgreSQL ........... SUCCESS [ 31.117 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: JIT Executor ........................ FAILURE [  0.006 s]<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: JIT Executor :: Common .............. SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: JIT Executor BPMN ................... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: JIT Executor DMN .................... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: JIT Executor Runner ................. SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests ................... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service ... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Common SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Data Index ..... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Data Index :: Common SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Common :: Quarkus SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus :: Management SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus :: Messaging SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus :: Knative Eventing SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Quarkus :: embedded SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Jobs Service :: Spring Boot SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service . SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Common SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Quarkus SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Audit UI  SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Trusty Service :: Spring Boot SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Data Index :: Quarkus SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Kogito Apps :: Integration Tests :: Data Index :: Spring Boot SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Runtime Tools Quarkus Extension - Parent ........... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Runtime Tools Quarkus Extension - Runtime .......... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] Runtime Tools Quarkus Extension - Deployment ....... SKIPPED<br/>[2023-06-23T08:47:40.348Z] [INFO] ------------------------------------------------------------------------<br/>[2023-06-23T08:47:40.348Z] [INFO] BUILD FAILURE<br/>[2023-06-23T08:47:40.348Z] [INFO] ------------------------------------------------------------------------<br/>[2023-06-23T08:47:40.348Z] [INFO] Total time:  59:42 min<br/>[2023-06-23T08:47:40.348Z] [INFO] Finished at: 2023-06-23T04:42:37-04:00<br/>[2023-06-23T08:47:40.348Z] [INFO] ------------------------------------------------------------------------<br/>[2023-06-23T08:47:40.348Z] ---------------------------------------------------<br/>[2023-06-23T08:47:40.348Z] constituent[0]: file:/opt/tools/apache-maven-3.8.7/conf/logging/<br/>[2023-06-23T08:47:40.348Z] constituent[1]: file:/opt/tools/apache-maven-3.8.7/lib/ext/takari-filemanager-0.8.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[2]: file:/opt/tools/apache-maven-3.8.7/lib/ext/takari-local-repository-0.11.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[3]: file:/opt/tools/apache-maven-3.8.7/lib/ext/takari-smart-builder-0.6.1.jar<br/>[2023-06-23T08:47:40.348Z] constituent[4]: file:/opt/tools/apache-maven-3.8.7/lib/maven-embedder-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[5]: file:/opt/tools/apache-maven-3.8.7/lib/maven-settings-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[6]: file:/opt/tools/apache-maven-3.8.7/lib/maven-settings-builder-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[7]: file:/opt/tools/apache-maven-3.8.7/lib/maven-plugin-api-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[8]: file:/opt/tools/apache-maven-3.8.7/lib/maven-model-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[9]: file:/opt/tools/apache-maven-3.8.7/lib/maven-model-builder-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[10]: file:/opt/tools/apache-maven-3.8.7/lib/maven-builder-support-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[11]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-api-1.6.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[12]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-util-1.6.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[13]: file:/opt/tools/apache-maven-3.8.7/lib/maven-shared-utils-3.3.4.jar<br/>[2023-06-23T08:47:40.348Z] constituent[14]: file:/opt/tools/apache-maven-3.8.7/lib/guice-4.2.2-no_aop.jar<br/>[2023-06-23T08:47:40.348Z] constituent[15]: file:/opt/tools/apache-maven-3.8.7/lib/guava-25.1-android.jar<br/>[2023-06-23T08:47:40.348Z] constituent[16]: file:/opt/tools/apache-maven-3.8.7/lib/javax.inject-1.jar<br/>[2023-06-23T08:47:40.348Z] constituent[17]: file:/opt/tools/apache-maven-3.8.7/lib/javax.annotation-api-1.2.jar<br/>[2023-06-23T08:47:40.348Z] constituent[18]: file:/opt/tools/apache-maven-3.8.7/lib/plexus-utils-3.3.1.jar<br/>[2023-06-23T08:47:40.348Z] constituent[19]: file:/opt/tools/apache-maven-3.8.7/lib/plexus-sec-dispatcher-2.0.jar<br/>[2023-06-23T08:47:40.348Z] constituent[20]: file:/opt/tools/apache-maven-3.8.7/lib/plexus-cipher-2.0.jar<br/>[2023-06-23T08:47:40.348Z] constituent[21]: file:/opt/tools/apache-maven-3.8.7/lib/plexus-interpolation-1.26.jar<br/>[2023-06-23T08:47:40.348Z] constituent[22]: file:/opt/tools/apache-maven-3.8.7/lib/slf4j-api-1.7.36.jar<br/>[2023-06-23T08:47:40.348Z] constituent[23]: file:/opt/tools/apache-maven-3.8.7/lib/commons-lang3-3.8.1.jar<br/>[2023-06-23T08:47:40.348Z] constituent[24]: file:/opt/tools/apache-maven-3.8.7/lib/maven-core-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[25]: file:/opt/tools/apache-maven-3.8.7/lib/maven-repository-metadata-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[26]: file:/opt/tools/apache-maven-3.8.7/lib/maven-artifact-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[27]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-provider-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[28]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-impl-1.6.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[29]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-spi-1.6.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[30]: file:/opt/tools/apache-maven-3.8.7/lib/org.eclipse.sisu.inject-0.3.5.jar<br/>[2023-06-23T08:47:40.348Z] constituent[31]: file:/opt/tools/apache-maven-3.8.7/lib/plexus-component-annotations-2.1.0.jar<br/>[2023-06-23T08:47:40.348Z] constituent[32]: file:/opt/tools/apache-maven-3.8.7/lib/maven-compat-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[33]: file:/opt/tools/apache-maven-3.8.7/lib/wagon-provider-api-3.5.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[34]: file:/opt/tools/apache-maven-3.8.7/lib/org.eclipse.sisu.plexus-0.3.5.jar<br/>[2023-06-23T08:47:40.348Z] constituent[35]: file:/opt/tools/apache-maven-3.8.7/lib/commons-cli-1.4.jar<br/>[2023-06-23T08:47:40.348Z] constituent[36]: file:/opt/tools/apache-maven-3.8.7/lib/wagon-http-3.5.3-shaded.jar<br/>[2023-06-23T08:47:40.348Z] constituent[37]: file:/opt/tools/apache-maven-3.8.7/lib/jcl-over-slf4j-1.7.36.jar<br/>[2023-06-23T08:47:40.348Z] constituent[38]: file:/opt/tools/apache-maven-3.8.7/lib/wagon-file-3.5.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[39]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-connector-basic-1.6.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[40]: file:/opt/tools/apache-maven-3.8.7/lib/maven-resolver-transport-wagon-1.6.3.jar<br/>[2023-06-23T08:47:40.348Z] constituent[41]: file:/opt/tools/apache-maven-3.8.7/lib/maven-slf4j-provider-3.8.7.jar<br/>[2023-06-23T08:47:40.348Z] constituent[42]: file:/opt/tools/apache-maven-3.8.7/lib/jansi-2.4.0.jar<br/>[2023-06-23T08:47:40.348Z] ---------------------------------------------------<br/>[2023-06-23T08:47:40.348Z] Exception in thread "main" javax.xml.parsers.FactoryConfigurationError: Provider for class javax.xml.parsers.SAXParserFactory cannot be created<br/>[2023-06-23T08:47:40.348Z] 	at java.xml/javax.xml.parsers.FactoryFinder.findServiceProvider(FactoryFinder.java:305)<br/>[2023-06-23T08:47:40.348Z] 	at java.xml/javax.xml.parsers.FactoryFinder.find(FactoryFinder.java:261)<br/>[2023-06-23T08:47:40.348Z] 	at java.xml/javax.xml.parsers.SAXParserFactory.newInstance(SAXParserFactory.java:147)<br/>[2023-06-23T08:47:40.348Z] 	at com.puppycrawl.tools.checkstyle.XmlLoader.<init>(XmlLoader.java:68)<br/>[2023-06-23T08:47:40.348Z] 	at com.puppycrawl.tools.checkstyle.ConfigurationLoader$InternalLoader.<init>(ConfigurationLoader.java:492)<br/>[2023-06-23T08:47:40.349Z] 	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.<init>(ConfigurationLoader.java:159)<br/>[2023-06-23T08:47:40.349Z] 	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:312)<br/>[2023-06-23T08:47:40.349Z] 	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:267)<br/>[2023-06-23T08:47:40.349Z] 	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:244)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.plugins.checkstyle.exec.DefaultCheckstyleExecutor.getConfiguration(DefaultCheckstyleExecutor.java:394)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.plugins.checkstyle.exec.DefaultCheckstyleExecutor.executeCheckstyle(DefaultCheckstyleExecutor.java:167)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.plugins.checkstyle.CheckstyleViolationCheckMojo.execute(CheckstyleViolationCheckMojo.java:545)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2(MojoExecutor.java:370)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute(MojoExecutor.java:351)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:215)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:171)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:163)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:298)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:960)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:293)<br/>[2023-06-23T08:47:40.349Z] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:196)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/>[2023-06-23T08:47:40.349Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:282)<br/>[2023-06-23T08:47:40.349Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:225)<br/>[2023-06-23T08:47:40.349Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:406)<br/>[2023-06-23T08:47:40.349Z] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:347)<br/>[2023-06-23T08:47:40.349Z] Caused by: java.lang.RuntimeException: Provider for class javax.xml.parsers.SAXParserFactory cannot be created<br/>[2023-06-23T08:47:40.349Z] 	at java.xml/javax.xml.parsers.FactoryFinder.findServiceProvider(FactoryFinder.java:302)<br/>[2023-06-23T08:47:40.349Z] 	... 35 more<br/>[2023-06-23T08:47:40.349Z] Caused by: java.util.ServiceConfigurationError: javax.xml.parsers.SAXParserFactory: Error accessing configuration file<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.parse(ServiceLoader.java:1173)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1206)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1221)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)<br/>[2023-06-23T08:47:40.349Z] 	at java.xml/javax.xml.parsers.FactoryFinder$1.run(FactoryFinder.java:287)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.security.AccessController.doPrivileged(Native Method)<br/>[2023-06-23T08:47:40.349Z] 	at java.xml/javax.xml.parsers.FactoryFinder.findServiceProvider(FactoryFinder.java:283)<br/>[2023-06-23T08:47:40.349Z] 	... 35 more<br/>[2023-06-23T08:47:40.349Z] Caused by: java.nio.file.NoSuchFileException: /home/jenkins/.m2/repository/xerces/xercesImpl/2.9.1/xercesImpl-2.9.1.jar<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.nio.file.Files.readAttributes(Files.java:1764)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1259)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:831)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.zip.ZipFile$CleanableResource$FinalizableResource.<init>(ZipFile.java:857)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:846)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:248)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:177)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.jar.JarFile.<init>(JarFile.java:350)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.net.www.protocol.jar.URLJarFile.<init>(URLJarFile.java:103)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.net.www.protocol.jar.URLJarFile.getJarFile(URLJarFile.java:72)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.net.www.protocol.jar.JarFileFactory.get(JarFileFactory.java:99)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:125)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/sun.net.www.protocol.jar.JarURLConnection.getInputStream(JarURLConnection.java:155)<br/>[2023-06-23T08:47:40.349Z] 	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.parse(ServiceLoader.java:1165)<br/>[2023-06-23T08:47:40.349Z] 	... 43 more<br/>[2023-06-23T08:47:40.349Z] [command]/usr/bin/docker system prune -f<br/>[2023-06-23T08:47:40.349Z] Deleted Containers:<br/>[2023-06-23T08:47:40.349Z] ad1b7cf8fcd144c32c86cf46727377efd1089961af2b4e7d9900b5aaa88d7081<br/>[2023-06-23T08:47:40.349Z] <br/>[2023-06-23T08:47:40.349Z] Deleted Networks:<br/>[2023-06-23T08:47:40.349Z] 3fa269b7-3832-4217-9cf9-f622c652a705<br/>[2023-06-23T08:47:40.349Z] <br/>[2023-06-23T08:47:40.349Z] Total reclaimed space: 63B<br/>[2023-06-23T08:47:40.349Z] [INFO] Execution summary for kiegroup/kogito-apps<br/>[2023-06-23T08:47:40.349Z] # [BEFORE] [kiegroup/kogito-apps] export INTEGRATION_BRANCH=<br/>[2023-06-23T08:47:40.349Z] [INFO] OK [Executed in 0.048261 ms]<br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z] # [BEFORE] [kiegroup/kogito-apps] bash -c "if [ ! -z '' ] && [ -f .ci/environments/update.sh ]; then .ci/environments/update.sh  ; fi"<br/>[2023-06-23T08:47:40.349Z] [INFO] OK [Executed in 5.254817 ms]<br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z] # [COMMANDS] [kiegroup/kogito-apps] export MVN_CMD=`bash -c "if [ '' = 'true' ]; then printf 'deploy  '; else printf 'install'; fi"`<br/>[2023-06-23T08:47:40.349Z] [INFO] OK [Executed in 5.705041 ms]<br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z] # [COMMANDS] [kiegroup/kogito-apps] mvn clean install  -s /home/jenkins/workspace/KIE/drools/main/pullrequest/drools.tests.downstream.kogito-apps@tmp/config10115339581107760923tmp -Dmaven.wagon.http.ssl.insecure=true -Dmaven.test.failure.ignore=true -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3   dependency:tree  -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B<br/>[2023-06-23T08:47:40.349Z] [INFO] NOT OK [Executed in 3584795.801801 ms]<br/>[2023-06-23T08:47:40.349Z] [ERROR] The process '/opt/tools/apache-maven-3.8.7/bin/mvn' failed with exit code 1<br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z] # [AFTER] [kiegroup/kogito-apps] docker system prune -f<br/>[2023-06-23T08:47:40.349Z] [INFO] OK [Executed in 234.689023 ms]<br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z] [INFO] kiegroup/kogito-apps failed. Won't execute remaining projects<br/>[2023-06-23T08:47:40.349Z] # Uploading artifacts<br/>[2023-06-23T08:47:40.349Z] [INFO] Will not upload any artifacts in CLI environment<br/>[2023-06-23T08:47:40.349Z]  <br/>[2023-06-23T08:47:40.349Z] [ERROR] Failed to execute commands for kiegroup/kogito-apps<br/>[2023-06-23T08:47:40.349Z] [ERROR] Failed to execute mvn clean install  -s /home/jenkins/workspace/KIE/drools/main/pullrequest/drools.tests.downstream.kogito-apps@tmp/config10115339581107760923tmp -Dmaven.wagon.http.ssl.insecure=true -Dmaven.test.failure.ignore=true -nsu -ntp -fae -e -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false -Dmaven.wagon.httpconnectionManager.ttlSeconds=120 -Dmaven.wagon.http.retryHandler.count=3   dependency:tree  -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B :<br/>[2023-06-23T08:47:40.349Z] [ERROR] The process '/opt/tools/apache-maven-3.8.7/bin/mvn' failed with exit code 1<br/>[Pipeline] }<br/>[Pipeline] // withCredentials<br/>[Pipeline] }<br/>[2023-06-23T08:47:40.497Z] Deleting 1 temporary files<br/>[Pipeline] // configFileProvider<br/>[Pipeline] }<br/>[Pipeline] // script<br/>Post stage<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-06-23T08:47:41.094Z] + find . -type d -name node_modules -exec rm -rf '{}' ';'<br/>[2023-06-23T08:47:44.353Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/packages/common/node_modules’: No such file or directory<br/>[2023-06-23T08:47:44.353Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/packages/components-common/node_modules’: No such file or directory<br/>[2023-06-23T08:47:44.353Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/packages/management-console-webapp/node_modules’: No such file or directory<br/>[2023-06-23T08:47:44.609Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/packages/patternfly-base/node_modules’: No such file or directory<br/>[2023-06-23T08:47:44.863Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/packages/runtime-tools-dev-ui-webapp/node_modules’: No such file or directory<br/>[2023-06-23T08:47:44.863Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/packages/trusty/node_modules’: No such file or directory<br/>[2023-06-23T08:47:45.125Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/node/node_modules’: No such file or directory<br/>[2023-06-23T08:48:00.100Z] find: ‘./bc/kiegroup_kogito-apps/ui-packages/node_modules’: No such file or directory<br/>[2023-06-23T08:48:00.100Z] + true<br/>[Pipeline] junit<br/>[2023-06-23T08:48:00.144Z] Recording test results<br/>[2023-06-23T08:48:03.236Z] [Checks API] No suitable checks publisher found.<br/>[Pipeline] archiveArtifacts<br/>[2023-06-23T08:48:03.254Z] Archiving artifacts<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>[2023-06-23T08:48:04.804Z] + rm -rf console.log<br/>[Pipeline] sh<br/>[2023-06-23T08:48:05.096Z] + wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/1486/consoleText<br/>[2023-06-23T08:48:05.096Z] + tail -n 300<br/>
</details>

tkobayas commented:
Thank you, @mariofusco  Now I fixed the test rules to meet your fixes in drools-ansible, removed `EqualityObjectStore` and reverted back to `IdentityObjectStore`. It works fine.

Btw, the added rule condition `protoPattern(eventVar).expr( thisPrototype(), Index.ConstraintType.EQUAL, controlVar3, prototypeField("event") ),` depends on `HashMapFactImpl.equals()` after deserialization, so I keep `HashMapFactImpl.uuid` and `equals`/`hashCode`. Is it okay?

kie-ci4 commented:
**(tests) - kogito-apps job** `#1490` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>


export BUILD_MVN_OPTS_CURRENT= 
build-chain build cross_pr -f 'https://raw.githubusercontent.com/kiegroup/drools/main/.ci/buildchain-config.yaml' -o 'bc' -p kiegroup/kogito-apps -u https://github.com/kiegroup/drools/pull/5344 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/1490/display/redirect

**Test results:**
- PASSED: 3096
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/drools/job/main/job/pullrequest/job/drools.tests.downstream.kogito-apps/1490/testReport/org.kie.kogito.index.service/KeycloakInfinispanIndexingServiceIT/testUnauthorizedUserAccess/">org.kie.kogito.index.service.KeycloakInfinispanIndexingServiceIT.testUnauthorizedUserAccess</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.kafka.client.deployment.DevServicesKafkaProcessor#startKafkaDevService threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at io.quarkus.kafka.client.deployment.DevServicesKafkaProcessor.startKafkaDevService(DevServicesKafkaProcessor.java:105)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:566)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:909)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:281)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2449)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1478)<br/> at java.base/java.lang.Thread.run(Thread.java:829)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:349)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:322)<br/> at io.quarkus.kafka.client.deployment.DevServicesKafkaProcessor.lambda$startKafka$5(DevServicesKafkaProcessor.java:236)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:369)<br/> at io.quarkus.kafka.client.deployment.DevServicesKafkaProcessor.startKafka(DevServicesKafkaProcessor.java:282)<br/> at io.quarkus.kafka.client.deployment.DevServicesKafkaProcessor.startKafkaDevService(DevServicesKafkaProcessor.java:95)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:334)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:542)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:344)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: java.lang.IllegalStateException: Container exited with code 1<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:514)<br/> ... 19 more<br/>
</details>

mariofusco commented:
> Btw, the added rule condition `protoPattern(eventVar).expr( thisPrototype(), Index.ConstraintType.EQUAL, controlVar3, prototypeField("event") ),` depends on `HashMapFactImpl.equals()` after deserialization, so I keep `HashMapFactImpl.uuid` and `equals`/`hashCode`. Is it okay?

Yes, that's ok.
